### PR TITLE
[react-native] Add hover props to PressableProps

### PR DESCRIPTION
**Pull request by Saadnajmi**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/facebook/react-native/pull/32405>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

I made a change that exposed new props on the Pressable component in React Native. This adds it to the types.
**EDIT:**
My change won't make it to React Native till RN 0.68, but I've backported it to older React Native macOS and React Native Windows releases. As such, I marked the props as platform specific.
Relevant PRs:
(RN macOS - 0.64) - https://github.com/microsoft/react-native-macos/pull/923
(macOS skipped RN 0.65, and went straight to 0.66)
(RN macOS - 0.66) - https://github.com/microsoft/react-native-macos/pull/884

(RN Windows - 0.64) - https://github.com/microsoft/react-native-windows/pull/9350
(RN Windows - 0.65) - https://github.com/microsoft/react-native-windows/pull/9349
(RN Windows - 0.66) - https://github.com/microsoft/react-native-windows/pull/9348
(RN Windows - 0.67) - https://github.com/microsoft/react-native-windows/pull/9347

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -500,6 +520,16 @@ export interface PressableProps extends AccessibilityProps, Omit<ViewProps, 'sty
      */
     cancelable?: null | boolean | undefined;
 
+    /**
+     * Duration to wait after hover in before calling `onHoverIn`.
+     */
+    delayHoverIn?: number,
```
Comment (ca057) [inline types/react-native/index.d.ts:526]: As this is a [maybe type](https://flow.org/en/docs/types/maybe/) according to the [source code](https://github.com/facebook/react-native/blob/main/Libraries/Pressability/Pressability.js#L58), I think it should be reflected here as 

```suggestion
    delayHoverIn?: number | null | undefined,
```

Same below :)

**Review by ca057 — COMMENTED** [review]
Looks good otherwise to me

**Review by gabritto — CHANGES_REQUESTED** [review]
please address the comment left by ca057, otherwise this looks good and we can then merge it.

File: types/react-native/index.d.ts (hunk 1 above)
Comment (Saadnajmi) [reply types/react-native/index.d.ts]: Sorry, missed this and just fixed it. Thanks for the catch!

**Comment by Saadnajmi** [comment]
@gabritto @ca057 It was mentioned to me that this specific change won't make it to a React-Native release till RN 0.67. Since the current typings target 0.66, I should perhaps hold off on this change until RN 0.67 is released.

**Comment by Saadnajmi** [comment]
@ca057 @gabritto I have no idea why the CI is failing here. Either of you might have any clues?

**Pull request closed without merging**